### DISCUSSION TITLE
How to recall the value of a MEM in a rule?

### DISCUSSION THREAD

**Discussion by mikip4** [arendst/Tasmota#15260] [Q&A] [answered]
Hello,
I'm working in a shutter and I want to save in mem10 open  duration
I save ShutterOpenDuration in mem10 whith:

ON mem10#state DO ShutterOpenDuration1 %value% ENDON

When power falls down I need to take mem10 value and set to ShutterOpenDuration1 
I use:

ON System#Init DO ShutterOpenDuration1 mem10 ENDON

but , not working.....

Can you help me?

Thanks a lot.

**Comment by sfromis** [comment, marked as answer]
Use `%mem10%` to have the value resolved.

**Reply by mikip4** [reply]
it works, thanks a lot

**Reply by mikip4** [reply]
sorry , another question....
how to capture in a var the ShutterPosition1 ?
Sorry, but I'm new programmong tasmota
Thanks
Miquel.

**Reply by mikip4** [reply]
I put
ON TuyaReceived#Data=55AA00070005010400010112 DO mem11 %ShutterPosition1% ENDON

but not working

**Reply by sfromis** [reply]
Right, can't work. You can use a JSON-based trigger in an extra rule to capture it, every time it appears. Not just when your TuyaReceived trigger fires. 
https://tasmota.github.io/docs/Rules/#rule-trigger

**Reply by ascillato** [reply]
`%ShutterPosition1%` is not a valid variable. The supported set of variables in a rule are only the ones specified in the docs at: https://tasmota.github.io/docs/Rules/#rule-variables

For storing `shutterposition` in a `MEM` or `VAR`, you will have to use another approach like:

`Rule1 ON Shutter1#Position DO VAR1 %value%" ENDON`

**Comment by mikip4** [comment]
Thanks a lot
Miquel
